### PR TITLE
Refactor paged lro methods

### PR DESCRIPTION
**Pull request by blowmage**
This PR allows for a method to be both paged and have the response wrapped. It is dependent on both googleapis/gax-ruby#190 and googleapis/gax-ruby#189.

### CODE REVIEW

File: shared/test/showcase/echo/wait_test.rb [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require "test_helper"
+require "google/showcase/v1alpha3/echo"
+
+class WaitTest < ShowcaseTest
```
Comment (blowmage) [inline shared/test/showcase/echo/wait_test.rb]: I think this could be broken out into its own PR.

Comment (dazuma) [reply shared/test/showcase/echo/wait_test.rb]: 🤷‍♀

File: shared/Gemfile [hunk 2]
```diff
@@ -6,7 +6,7 @@ gem "gapic-generator", path: "../gapic-generator"
 gem "gapic-generator-cloud", path: "../gapic-generator-cloud"
 gem "googleapis-common-protos", [">= 1.3.9", "< 2.0"]
 gem "googleapis-common-protos-types", [">= 1.0.4", "< 2.0"]
-gem "google-gax", github: "googleapis/gax-ruby"
+gem "google-gax", github: "blowmage/gax-ruby", branch: "2.0/update-operation-constructor"
```
Comment (dazuma) [inline shared/Gemfile]: I assume this will get reverted.

Comment (blowmage) [reply shared/Gemfile]: Yes, as soon as googleapis/gax-ruby#189 is merged.

Comment (blowmage) [reply shared/Gemfile]: Reverted.

**Comment by blowmage** [comment]
I updated to the latest google-style to get a passing build.
